### ISSUE TITLE
Error from jsonmark.js when using latest version

### ISSUE THREAD

**Issue by josefsalyer** [julianxhokaxhiu/gitbook-plugin-summary#2]
When running gitbook serve I receive this error:

```
/path/to/node_modules/jsonmark/jsonmark.js:32
          headerObj.body.push(row)
                   ^

TypeError: Cannot read property 'body' of undefined
    at /path/to/docs/node_modules/jsonmark/jsonmark.js:32:20
    at Array.forEach (native)
    at Object._parseContent (/path/to/docs/node_modules/jsonmark/jsonmark.js:17:12)
    at Object.parse (/path/to/docs/node_modules/jsonmark/jsonmark.js:63:27)
    at files.forEach (/path/to/docs/node_modules/gitbook-plugin-summary/index.js:32:39)
    at Array.forEach (native)
    at glob (/path/to/docs/node_modules/gitbook-plugin-summary/index.js:31:17)
    at f (/path/to/docs/node_modules/once/once.js:25:25)
    at Glob.<anonymous> (/path/to/docs/node_modules/glob/glob.js:151:7)
    at emitOne (events.js:96:13)
```

package.json
```
{
  "name": "docs",
  "version": "1.0.0",
  "description": "a gitbook",
  "dependencies": {
    "gitbook": "^3.2.3",
    "gitbook-plugin-search-plus-mod": "^1.0.6",
    "gitbook-plugin-summary": "^0.1.2",
    "gitbook-plugin-tags": "^1.5.1",
    "svgexport": "^0.3.2"
  },
  "devDependencies": {},
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}
```

book.json
```
{
  "title":"title",
  "author": "author",
  "description": "description",
  "plugins": [
    "tags", "-lunr", "-search", "search-plus-mod", "summary"
  ],
  "pluginsConfig": {
    "autocover": {
      "font": {
        "size": null,
        "family": "Impact",
        "color": "#FFF"
      },
      "size": {
        "w": 1800,
        "h": 2360
      },
      "background": {
        "color": "#09F"
      }
    }
  },
  "pdf": {
    "fontFamily":"Montserrat"
  },
  "variables": {
    "title": "A DevOps Playbook"
  },
  "ignores": [ "node_modules", "testing/penetration_testing.md" ]

}
```

**Comment by julianxhokaxhiu** [comment]
Can you please try version 0.1.3?

**Comment by tbouron** [comment]
Hi @julianxhokaxhiu. I get the same error with version `0.1.3`

**Comment by julianxhokaxhiu** [comment]
@tbouron and @josefsalyer can you provide a `tree` view of your folder where the `book.json` file is present?

Thank you.

**Comment by tbouron** [comment]
There you go @julianxhokaxhiu 

<img width="299" alt="screen shot 2017-09-29 at 13 26 32" src="https://user-images.githubusercontent.com/2082759/31014172-dc25c70c-a519-11e7-9e82-b98f9fda5d5c.png">

**Comment by julianxhokaxhiu** [comment]
Could it be that the `.DS_Store` is the issue?

Can you place a `console.log(files)` before this line https://github.com/julianxhokaxhiu/gitbook-plugin-summary/blob/master/index.js#L30 ? Please paste the output here. Thank you.

**Comment by tbouron** [comment]
Here is the output @julianxhokaxhiu 

```
Live reload server started on port: 35729
Press CTRL+C to quit ...

info: 8 plugins are installed
info: loading plugin "summary"... OK
info: loading plugin "livereload"... OK
info: loading plugin "highlight"... OK
info: loading plugin "search"... OK
info: loading plugin "lunr"... OK
info: loading plugin "sharing"... OK
info: loading plugin "fontsettings"... OK
info: loading plugin "theme-default"... OK
info: found 4 pages
info: found 2 asset files
[ 'download/index.md', 'download/verify.md' ]
/Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/jsonmark/jsonmark.js:32
          headerObj.body.push(row)
                   ^

TypeError: Cannot read property 'body' of undefined
    at /Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/jsonmark/jsonmark.js:32:20
    at Array.forEach (<anonymous>)
    at Object._parseContent (/Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/jsonmark/jsonmark.js:17:12)
    at Object.parse (/Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/jsonmark/jsonmark.js:63:27)
    at files.forEach (/Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/gitbook-plugin-summary/index.js:34:39)
    at Array.forEach (<anonymous>)
    at glob (/Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/gitbook-plugin-summary/index.js:33:17)
    at f (/Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/once/once.js:25:25)
    at Glob.<anonymous> (/Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/glob/glob.js:151:7)
    at emitOne (events.js:115:13)
```

**Comment by julianxhokaxhiu** [comment]
Sorry wrong line, Can you `console.log(filePath)` before this line https://github.com/julianxhokaxhiu/gitbook-plugin-summary/blob/master/index.js#L33 ? As always provide some logs. Thank you!

Also please provide the content of the two md files in your download folder. Are they empty? ( 0 byte files )

**Comment by tbouron** [comment]
My 2 files are not empty, they have [standard markdown + html syntax](https://github.com/apache/brooklyn-docs/tree/master/website/download)

Here is the new output:
```
Live reload server started on port: 35729
Press CTRL+C to quit ...

info: 8 plugins are installed
info: loading plugin "summary"... OK
info: loading plugin "livereload"... OK
info: loading plugin "highlight"... OK
info: loading plugin "search"... OK
info: loading plugin "lunr"... OK
info: loading plugin "sharing"... OK
info: loading plugin "fontsettings"... OK
info: loading plugin "theme-default"... OK
info: found 4 pages
info: found 2 asset files
download/index.md
/Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/jsonmark/jsonmark.js:32
          headerObj.body.push(row)
                   ^

TypeError: Cannot read property 'body' of undefined
    at /Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/jsonmark/jsonmark.js:32:20
    at Array.forEach (<anonymous>)
    at Object._parseContent (/Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/jsonmark/jsonmark.js:17:12)
    at Object.parse (/Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/jsonmark/jsonmark.js:63:27)
    at files.forEach (/Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/gitbook-plugin-summary/index.js:34:39)
    at Array.forEach (<anonymous>)
    at glob (/Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/gitbook-plugin-summary/index.js:32:17)
    at f (/Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/once/once.js:25:25)
    at Glob.<anonymous> (/Users/thomasbouron/repositories/jekyll-tests/gitbook/node_modules/glob/glob.js:151:7)
    at emitOne (events.js:115:13)
```

**Comment by julianxhokaxhiu** [comment]
Ok found the issue, it seems that that your markup that is not supported by this library is triggering an uninitialization on the library code, which is unexpected at that point. I will see if I can find a better library to parse Markdown, so I can maybe support your syntax too.

**Comment by tbouron** [comment]
Ok cool, thanks @julianxhokaxhiu

**Comment by julianxhokaxhiu** [comment]
@tbouron I just published 0.1.4. Migrated to https://www.npmjs.com/package/markdown-parser but the final result is identically the same.

It should work now, can you please try it?
